pubtools/pulplib/_impl/model/frozenlist.py

```diff
@@ -0,0 +1,35 @@
+class FrozenList(list):
```

Comment:
What the benefit of having custom class instead of tuple?

Reply:
I just checked the doc of atrs, for 'default' value for an attr.ib, it says:
`If the value is an instance of Factory, its callable will be used to construct a new value (useful for mutable data types like lists or dicts).`

So, if it's a Frozen list, which inherits from list, then a new FronzenList object will be constructed, but if it's a tuple, it won't.

Reply:
I wrote explanation for that in https://github.com/release-engineering/pubtools-pulplib/pull/49/commits/75790f140edc0f4fe1a330bcf407cee8fcce0e37 commit message.  In summary: the previous solution *was* to use tuple but it has usability issues because developers choose lists as the default container and tuples aren't as compatible with them as it may seem at first glance.  For instance `(1, 2, 3) == [1, 2, 3]` isn't true.

That means, if it continued using tuples, developers have to get in the habit of remembering where they need to write a tuple and where they need to write a list, or writing tuples everywhere (which is a tough ask), or running into weird bugs.